File: uires/src/main/java/com/pandaq/uires/msgwindow/SnackerConfig.java
package com.pandaq.uires.msgwindow;

/**
 * Created by huxinyu on 2018/12/3.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :default config for Snacker & Toaster
 */
public class SnackerConfig {

    public static int COLOR_NULL = -1;
    private static SnackerConfig sDefault;
    private int msgColor = COLOR_NULL;
    private int actionColor = COLOR_NULL;
    private int backgroundColor = COLOR_NULL;

    public static SnackerConfig getDefault() {
        if (sDefault == null) {
            synchronized (SnackerConfig.class) {
                sDefault = new SnackerConfig();
            }
        }
        return sDefault;
    }

    public int getMsgColor() {
        return msgColor;
    }

    public SnackerConfig setMsgColor(int msgColor) {
        this.msgColor = msgColor;
        return this;
    }

    public int getActionColor() {
        return actionColor;
    }

    public SnackerConfig setActionColor(int actionColor) {
        this.actionColor = actionColor;
        return this;
    }

    public int getBackgroundColor() {
        return backgroundColor;
    }

    public SnackerConfig setBackgroundColor(int backgroundColor) {
        this.backgroundColor = backgroundColor;
        return this;
    }
}

File: uires/src/main/java/com/pandaq/uires/configs/SnackerConfig.kt
package com.pandaq.uires.configs

import androidx.annotation.ColorInt

/**
 * Created by huxinyu on 2019/8/8.
 * Email : panda.h@foxmail.com
 * Description :
 */
class SnackerConfig {

    companion object {
        const val COLOR_NULL = -1
        private val config by lazy {
            SnackerConfig()
        }

        fun get(): SnackerConfig = config
    }

    private var msgColor = COLOR_NULL
    private var actionColor = COLOR_NULL
    private var backgroundColor = COLOR_NULL

    fun setMsgColor(@ColorInt color: Int): SnackerConfig {
        this.msgColor = color
        return this
    }

    fun setActionColor(@ColorInt color: Int): SnackerConfig {
        this.actionColor = color
        return this
    }

    fun setBackgroundColor(@ColorInt color: Int): SnackerConfig {
        this.backgroundColor = color
        return this
    }

    fun getMsgColor(): Int {
        return msgColor
    }

    fun getActionColor(): Int {
        return actionColor
    }

    fun getBackgroundColor(): Int {
        return backgroundColor
    }
}